components/segmenter/src/complex/mod.rs

```diff
@@ -322,10 +363,10 @@ impl ComplexPayloads {
 
 fn try_load<M: KeyedDataMarker, P: DataProvider<M> + ?Sized>(
     provider: &P,
-    locale: LanguageIdentifier,
+    model: &'static str,
 ) -> Result<Option<DataPayload<M>>, DataError> {
     match provider.load(DataRequest {
-        locale: &locale.into(),
+        key_attributes: &model.parse().unwrap(),
```

Comment:
Nit: We shouldn't need to `unwrap` anything; I'm surprised Clippy isn't complaining since this is library code.

Reply:
This is an infallible FromStr, which clippy apparently realises. The whole DataKeyAttributes construction will be overhauled anyway once we are clearer about requirements. Ideally we'd construct from static str here.